accounts/abi/abi.go

```diff
@@ -222,6 +222,20 @@ func (abi *ABI) EventByID(topic common.Hash) (*Event, error) {
 	return nil, fmt.Errorf("no event with id: %#x", topic.Hex())
 }
 
+// ErrorById looks up an error by the 4-byte id,
+// returns nil if none found.
+func (abi *ABI) ErrorById(sigdata []byte) (*Error, error) {
+	if len(sigdata) < 4 {
+		return nil, fmt.Errorf("data too short (%d bytes) for abi error lookup", len(sigdata))
+	}
+	for _, errABI := range abi.Errors {
+		if bytes.Equal(errABI.ID, sigdata[:4]) {
```

Comment:
I wonder why you changed the `ID` field to `[]byte`. Might be better to:

- change the type of the `sigdata` parameter to `[4]byte`
- do the check like bytes.Equal(errABI.ID[:4], sigdata[:])`